jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ReHandlingErrorProcessor.java

```diff
@@ -0,0 +1,142 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.ExceptionUtil;
+import org.eclipse.jetty.util.URIUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An ErrorProcessor that can re-handle a request at an error page location.
+ */
+public abstract class ReHandlingErrorProcessor extends ErrorProcessor
+{
+    private static final Logger LOG = LoggerFactory.getLogger(ReHandlingErrorProcessor.class);
+
+    private final Handler _handler;
+
+    protected ReHandlingErrorProcessor(Handler handler)
```

Comment:
I wonder whether we should be more strict here and only accept `Server`.

Otherwise this class can be configured with a `ContextHandler("/foo")`, but the rehandling code in `generateResponse()` rebuild the URI with the current request `contextPath` which may be different.

Reply:
ContextHandlers can have their own ErrorProcessors, so in that case, we are only rewriting the pathInContext and redispatching back to the same context.
I guess there are also strange use-cases that might have errors handled in an entirely different context or maybe just a handler with no context at all.

Reply:
@sbordet I added the context error test case.  OK?